src/cljs/main/broadfcui/common/method/config_io.cljs

```diff
@@ -21,41 +22,63 @@
 
 (react/defc IOTables
   {:start-editing
-   (fn [{:keys [props state locals]}]
-     (swap! locals utils/deep-merge (:values props))
-     (swap! state assoc :editing? true))
+   (fn [{:keys [props state]}]
+     (let [{:keys [outputs]} @state]
+       (swap! state assoc :editing? true)
+       (swap! state utils/deep-merge (:values props))
+       (swap! state assoc :original-outputs outputs)))
    :cancel-editing
    (fn [{:keys [state]}]
-     (swap! state dissoc :editing?))
+     (let [{:keys [original-outputs]} @state]
+       (swap! state dissoc :editing?)
+       (swap! state assoc :outputs (utils/log original-outputs))))
    :save
-   (fn [{:keys [props state locals]}]
+   (fn [{:keys [props state]}]
      (swap! state dissoc :editing?)
-     {:inputs (select-keys (:inputs @locals) (map (comp keyword :name) (:inputs (:inputs-outputs props))))
-      :outputs (select-keys (:outputs @locals) (map (comp keyword :name) (:outputs (:inputs-outputs props))))})
+     {:inputs (select-keys (:inputs @state) (map (comp keyword :name) (:inputs (:inputs-outputs props))))
+      :outputs (select-keys (:outputs @state) (map (comp keyword :name) (:outputs (:inputs-outputs props))))})
+   :-get-defaultable-outputs
+   (fn [{:keys [props state]}]
+     (let [{:keys [inputs-outputs]} props
+           {:keys [outputs]} @state]
+       (filter #(string/blank? ((keyword (:name %)) outputs))
+               (:outputs inputs-outputs))))
+   :-add-default-outputs
+   (fn [{:keys [props state this]}]
+     (let [{:keys [begin-editing]} props
+           {:keys [outputs editing?]} @state
+           defaultable-outputs (this :-get-defaultable-outputs)
+           new-outputs (merge outputs (into {} (map (fn [output] {(keyword (:name output)) (str "this." (last (string/split (:name output) ".")))}) defaultable-outputs)))]
```

Comment:
This line is pretty dense. Some newlines would help, and also maybe some threading macros. Maybe:
```clojure
(->> defaultable-outputs
     (map (fn [{:keys [name]}] [(keyword name) (str "this." (-> name (string/split ".") last))]))
     (into {})
     (merge outputs))